### ISSUE TITLE
Validation Page Error

### ISSUE THREAD

**Issue by jctomlin** [tagyoureit/hubitat-luxor#2]
When trying to add controller, get the attached error. No “Back” button to assign IP address for controller. Help?

**Comment by tagyoureit** [comment]
Which SmartThings app are you using?  I haven't touched this code in 3 years and it's working for me on the old platform still.  Assuming you are on the new app.  If so, it will take me a while to get to this.

**Comment by jctomlin** [comment]
Yeah, I was trying to install it with the new SmartThings app--I tried the classic version and it worked fine.  No need to spend time on it.  It's working great.

**Comment by jctomlin** [comment]
One question:
When I add/remove themes in Luxor, how do I get them to resync in ST?
Thanks for your help. 


Sent from Yahoo Mail for iPhone


On Monday, July 6, 2020, 11:16 PM, tagyoureit <notifications@github.com> wrote:




Which SmartThings app are you using? I haven't touched this code in 3 years and it's working for me on the old platform still. Assuming you are on the new app. If so, it will take me a while to get to this.

**Comment by tagyoureit** [comment]
Go back through the setup process again and it should pick up the new themes.

**Comment by jctomlin** [comment]
Yeah I tried that. No luck. 🤷‍♂️


Sent from Yahoo Mail for iPhone


On Tuesday, July 7, 2020, 8:34 PM, tagyoureit <notifications@github.com> wrote:




Go back through the setup process again and it should pick up the new themes.

**Comment by tagyoureit** [comment]
Hi, I just pushed up some code changes.  I also tested this and it was working for me.  If you don't have success look in the logs and see what it says there.

**Comment by jctomlin** [comment]
I’ll give it a shot. 




On Tuesday, August 4, 2020, 9:52 PM, tagyoureit <notifications@github.com> wrote:




Hi, I just pushed up some code changes. I also tested this and it was working for me. If you don't have success look in the logs and see what it says there.

**Comment by jctomlin** [comment]
I’m able to add the controller and all works fine through the new ST app. However, when I rename a theme in Luxor or add one, it doest show up in ST. Even if I go through the Setup screen again. Thoughts?  I haven’t yet tried to remove it from ST then re-add it. 




On Wednesday, August 5, 2020, 7:30 PM, Jason Tomlin <jason_tomlin@yahoo.com> wrote:

I’ll give it a shot. 




On Tuesday, August 4, 2020, 9:52 PM, tagyoureit <notifications@github.com> wrote:




Hi, I just pushed up some code changes. I also tested this and it was working for me. If you don't have success look in the logs and see what it says there.

**Comment by tagyoureit** [comment]
Can you go to the parent device (Luxor xyz Controller) and set the logging to "trace".  Then open up the logs and try to rename one of the themes.  Go back to the SmartApp and let it refind/reconfigure the parent controller.  Cut/paste the output here and I'll see what it's picking up.  Also, tell me which one you renamed and the old/new names.

**Comment by jctomlin** [comment]
I'm afraid I don't know how to do that first part.  Is it in the device handler?  But I do know how to look at the logs.
go to the parent device (Luxor xyz Controller) and set the logging to "trace"
    On Thursday, August 6, 2020, 09:02:12 PM PDT, tagyoureit <notifications@github.com> wrote:  
 
 


Can you go to the parent device (Luxor xyz Controller) and set the logging to "trace". Then open up the logs and try to rename one of the themes. Cut/paste the output here and I'll see what it's picking up. Also, tell me which one you renamed and the old/new names.

**Comment by tagyoureit** [comment]
First, pull the latest code for the app and device handlers.  I just pushed up a few fixes.

To set the logging level, you can do it one of two ways:
### In your SmartThings app
1. Go to My Home/Things
2. Click on the name (not button) of 'Luxor xyz Controller'
3. Click the gear icon in the upper right
4. Select "Trace" logging level and click save (upper right)

### In a browser
1.  Log in to your Hub (for NA: https://graph-na04-useast2.api.smartthings.com/)
2.  Click My Devices
3. Click on the name 'Luxor xyz Controller'
4. Click edit next to Preferences
5. Type in "Trace" (must be capital T) 
6. Click save

# To capture logs
1. Log into your hub
2. Click on "Live Logging"
3. There could be lots of activity here; once you "Luxor xyz Controller" appear in the list of devices you should click on it to filter to just this controller
4. Go back to your Luxor and add a new theme, change a name, etc.  Every time this app updates it's status it checks to see if the right "children" (lights/themes) are installed and if they need to be update.
5. Go back to the the 'Luxor xyz Controller' and now hit the button to turn it on/off.  You'll see a whole lot of stuff in the logs.  Copy and paste it all (probably into a text file) and attach it here. 

I just added a "New theme" and it shows up like this:
`
299f1019-... 7:46:06 AM: info Creating Luxor Theme #:0, name New theme
299f1019-... 7:46:08 AM: info THEME 6. Light Theme New theme Added`

Then I changed the name from "New theme" to "New name" and you'll see this in the logs:
`299f1019-... 7:52:12 AM: debug theme [Name:New name, OnOff:0, ThemeIndex:0]`
It isn't specifically called out that there is a name change but it is and I see a "New name" theme in my list of devices.

**Comment by jctomlin** [comment]
OK--I've updated all the code.  I'm able to turn on/off the controller, but cannot select a theme--any of them.  Here's the output:

495148bb-29af-4117-86cd-543d2f248742 9:16:06 AM: error java.lang.NullPointerException: Cannot get property 'device' on null object @line 88 (sendCommandToController)
495148bb-29af-4117-86cd-543d2f248742 9:16:06 AM: debug Executing 'push' on Blue and Purple

**Comment by jctomlin** [comment]
I appreciate your help on this, can I get a rollback until you get a look at this? I’m happy choosing themes over renaming them for now. Thanks.

**Comment by tagyoureit** [comment]
I'm not able to replicate this... can you make sure you've updated and published the smart app and child devices?

Here is my output.
```
uuid 10:48:04 PM: debug Sending /IlluminateTheme.json to controller
POST /IlluminateTheme.json HTTP/1.1
Accept: */*
User-Agent: Linux UPnP/1.0 SmartThings
HOST: 10.0.0.13:80
Content-Type: application/json
Content-Length: 26

{"ThemeIndex":1,"OnOff":1}
uuid 10:48:04 PM: debug Executing 'push' on Test theme
```

**Comment by jctomlin** [comment]
OK--I was able to get it going.  I uninstalled the smartapp and reinstalled.  One thing to note, during the setup (with the new ST app), it could not find the controller at IP xxx.xxx.xxx.xxx.  There was nowhere I could enter the IP of the controller.  So I set up the app in the classic ST app and it installed fine and imported everything.  For the renaming--all works good now.  Thanks for this!

**Comment by garyhooper** [comment]
I am using the new app, as the classic app is now defunct. There is no place to enter the IP address of my Luxor controller.

![Luxor](https://user-images.githubusercontent.com/11620531/120601106-8899ba00-c417-11eb-8685-dc975803affd.jpg)

**Comment by tagyoureit** [comment]
I'm not really using Smartthings anymore and don't have any active plans to port this app (or my others) over to the new ST platform.  I do recommend Hubitat and the app/drivers are fully supported there.

**Comment by garyhooper** [comment]
Thanks @tagyoureit. Are you happy with Hubitat as a user?

**Comment by jctomlin** [comment]
I'm able to use it with the new ST app. I had to fix the IP addy on my Luxor controller to what it was in the old app.
Yeah, I'd also like to know how you like Hubitat....
    On Thursday, June 3, 2021, 12:32:52 PM PDT, Gary Hooper ***@***.***> wrote:  
 
 


Thanks @tagyoureit. Are you happy with Hubitat as a user?

**Comment by tagyoureit** [comment]
Yes, Hubitat is great to work with.  I do recommend it.
